Data Structure/Heap/Heap.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+function MaxHeapify (&$data, $heapSize, $index) {
+    $left = $index * 2 + 1;
+    $right = $index * 2 + 2;
+    $largest = 0;
+
+    if ($left < $heapSize && $data[$left] > $data[$index]) {
+        $largest = $left;
+    } else {
+        $largest = $index;
+    }
+
+    if ($right < $heapSize && $data[$right] > $data[$largest]) {
+        $largest = $right;
+    }
+
+    if ($largest !== $index) {
+        $temp = $data[$index];
+        $data[$index] = $data[$largest];
+        $data[$largest] = $temp;
+
+        MaxHeapify($data, $heapSize, $largest);
+    }
+}
+
+function HeapSort(&$data, $count) {
+    $heapSize = $count;
+
+    for ($p = $heapSize / 2 - 1 ; $p >= 0; $p--) {
+        MaxHeapify($data, $heapSize, $p);
+    }
+
+    for ($i = $count - 1; $i > 0; $i--) {
+        $temp = $data[$i];
+        $data[$i] = $data[0];
```

Comment:
Why don't reuse Heapsort and output k Largest Number in here?
Please try to improve faster by out put k number in here and then return the function.